src/main/java/frc/team555/FootballShooter/RobotLauncher.java

```diff
@@ -0,0 +1,88 @@
+package frc.team555.FootballShooter;
+
+
+import frc.robot.core.RobotLikeComponent;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+
+import edu.wpi.first.wpilibj.Compressor;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.buttons.JoystickButton;
+
+
+public class RobotLauncher implements RobotLikeComponent {
```

Comment:
Is this class still needed? I'm just wondering why there is a Launcher.java and a RobotLauncher.java that seemingly do the same things.

Reply:
I assume that one can replace the other, but I don't have an opinion at this time which is preferable.  I did remove a reference in one to c22a2c092a03335b3af75b7c01d83310fe07bc96 but I didn't think it prudent to remove one or the other class completely until the respective authors - or some third party - chose which should be used, or if they should be merged (though of course nothing is ever completely gone in `git` world).

I do note that one - `src/main/java/frc/robot/core/Launcher.java` - is retaining a link to a `Robot` object.  That makes me nervous as it suggests a bidirectional dependency - never a good thing.  But I don't know how the author expects to use this, so...I don't know.

I do know that one would typically avoid such a dependency by creating some third entity - typically an interface in java - on which both `Robot` and `Launcher` would somehow be dependent.  For example:
- `Robot` might implement that interface, 
- `Launcher` might take and store a reference to that interface, and
- `Launcher` would invoke some behaviour via the interface's public set of methods

This should look familiar.  It is the same approach as that which I took in turning the Robot object into a collection of RobotLikeComponent objects.  Each is an exploitation/example of the Dependency Inversion Principle: https://www.javacodegeeks.com/2018/02/solid-principles-dependency-inversion-principle.html (though, technically, the interface(s) shared by the container/consumer and the components/providers should be in a separate package to achieve maximum independence).

From a workflow perspective, perhaps the work involved in choosing a specific Launcher class - or the merging of the two - should be a separate effort and thus occurring in a separate branch.